### ISSUE TITLE
Override ignorePaths

### ISSUE THREAD

**Issue by pltrant** [streetsidesoftware/vscode-spell-checker#1808]
Is it possible to override the ignorePaths global setting at the folder level?  Specifically, the default setting has:

<pre>
    "cSpell.ignorePaths": [
        "package-lock.json",
        "node_modules",
        "vscode-extension",
        ".git/objects",
        ".vscode-insiders",
        ".vscode"
    ]
</pre>

I want to remove `.vscode` at the folder level via the `cspell.json`.  However, no matter what settings I use in the `cspell.json`, it doesn't remove it.  I even tried a null entry, thinking it would override all the settings (which is fine too).

<pre>
{
    "version": "0.2",
    "ignorePaths": []
}
</pre>

The other problem is, even when I remove `.vscode` at the global settings, it still says the file was not spell checked because "The file path is excluded in settings." (as opposed to "File excluded by ".vscode" in VSCode-Config".  Having researched some similar issues, I also verified it's not due to my `search.exclude` settings, which is just:

<pre>
**/node-modules
**/bower_components
**/*.code-search
</pre>

Some additional context for the problem and what I'm trying to solve.  I have an extension that downloads some files into the same folder as the extension itself.  I want to spellcheck those files.  There's a `cspell.json` in the same folder, but I cannot get it to spellcheck any of the files due to the above issue ("file path is excluded in settings").  Since it's all working from the extension directory, it has a path of `C:\Users\<name>\.vscode\extensions\...`, so I thought I could just remove the `.vscode` exclusion, but that does not appear to be enough.  Any ideas of other settings I could try for my local `cspell.json` to enable it work?

**Comment by Jason3S** [comment]
@pltrant,

Thank you for the detailed report. This is an issue with how the VS Code extension and the `cspell` library interact. In your VS Code settings, set:

```
"cSpell.ignorePaths": []
```

That will override the defaults.

**tldr;**

As cspell configurations are read, they are merged to create a final configuration used to spell check a file. `ignorePaths` is also merged. By default, the `cspell` library does not have any `ignorePaths` defined. But since the extension defines some by default, they are merged with `ignorePaths` found in your `cspell.json`.

This issue, does expose a problem with the merge logic, since it is not possible to remove an `ignorePath` from the list.

**Comment by pltrant** [comment]
Thanks for the response.  Unfortunately, even with `"cSpell.ignorePaths": []` in the VS Code setting, it simply changes the message from "[file] is NOT spell checked.  File excluded by ".vscode" in VSCode-Config" to "[file] is NOT spell checked.  The file path is excluded in settings."

**Comment by Jason3S** [comment]
@pltrant,

I'll take a look. You might need to restart VS Code. It has a bug where old settings are cached when a new setting is added. Especially if you didn't clear the setting via the UI.

**Comment by Jason3S** [comment]
@pltrant,

A bit more information is needed.

- How are you opening the files?
   When I do `code C:\Users\<name>\.vscode\extensions\`, it spell checks the files. 
- Do you have Spell Check Only Workspace Files set?
   ![image](https://user-images.githubusercontent.com/3740137/157065347-542fe50e-a93a-44ce-94e6-d95106f46b55.png)
- Or did you add any files to `files`?
   ![image](https://user-images.githubusercontent.com/3740137/157065787-8af65168-7ec5-4ce1-91b8-c80556909f12.png)

**Comment by Jason3S** [comment]
@pltrant,

Did you click on the ![image](https://user-images.githubusercontent.com/3740137/157068433-5655a477-d433-4657-a0bc-238053d3963a.png)?

![image](https://user-images.githubusercontent.com/3740137/157068385-e28751b9-4fca-4128-9e83-ac439b1b9fd6.png)

It should show more information:
![image](https://user-images.githubusercontent.com/3740137/157068587-593fa3c4-bead-4458-9712-970be27ed80a.png)

**Comment by pltrant** [comment]
The file is downloaded into the same directory as the extension, then opened in the editor.  The "Spell Check Only Workspace Files" setting was not enabled, but I tried toggling it a few times to potentially resolve the issue, as it does appear to be related as when I click on the Spell button in the Status Bar, it then shows the reason of "File is NOT spell checked because it is not in the workspace".

The file extension and type (language ID) is `gsl` and I have this in the `cspell.json`:

<pre>
    "files": [
        "*.gsl"
    ],
    "enableFiletypes": [
        "gsl"
    ]
 </pre>
 
 I also tried duplicating the same in the VS Code setting without any difference.  I tried closing and restarting VSC to clear any potential cache issues with the setting.

![image](https://user-images.githubusercontent.com/9040032/157080233-5cf6944c-3169-4d38-96ce-d5daa5993f8e.png)

![image](https://user-images.githubusercontent.com/9040032/157080096-e72f2753-12e1-456a-a4cd-48201f51d20e.png)

**Comment by Jason3S** [comment]
@pltrant,

`"files": ["*.gsl"]` tells the spell checker to only check files in the same directory as the `cspell.json` file. Please try `"files": ["**/*.gsl"]`.

**Comment by pltrant** [comment]
Actually, after some additional testing, that did not resolve the issue.  I actually have the file (s00001.gsl) in the same directory as the `cspell.json`:

![image](https://user-images.githubusercontent.com/9040032/157766812-e9fb26ae-f709-4245-a631-a16c7cf47991.png)

![image](https://user-images.githubusercontent.com/9040032/157767015-c2c45981-74c2-4bf2-8f97-c9906c83d7f4.png)

![image](https://user-images.githubusercontent.com/9040032/157767063-06c43e78-b2d6-4742-b34f-17e7b23d2996.png)

Even if I add `"cSpell.spellCheckOnlyWorkspaceFiles": false` to my VSC Settings, it doesn't change the result (including after closing/restarting VSC).

**Comment by Jason3S** [comment]
@pltrant,

A few questions:
- How are you opening the files?
   Please try opening the `.vscode/extensions/gsl` directory in VSCode and see what happens.
- Do you need the `files` section? It is used as a way to limit which files are checked.

Logic:

```
| - root
    | - projects/project A
        | - cspell.json (files: **)
        | - README.md
    | - .vscode/extensions/gsl
        | - cspell.json (files: *.gsl)
        | - s0000.gsl
        | - more .gsl files.
        | - README.md (not checked)
```

If you have the `project A` workspace open, `files` prevents spell checking any files outside of it.

**Comment by pltrant** [comment]
Thanks again, @Jason3S.  The files are downloaded from a server, then opened in the editor.  Per your tip, I discovered that if I opened the folder, the spellcheck then worked.  It appears to somewhat require a Folder or Workspace to be opened and not individual files.  However, if I removed the `files` setting, then it also worked without needing to open it at the Folder or Workspace level, but it seems odd, as the `files` setting should still work under this scenario because the the setting and file extension aligned.  But you're right, I don't need to limit the checks just to those types of files, so at least on my end, the issue is resolved.  Much appreciate the responses and guidance!
